x-pack/plugins/security_solution/server/endpoint/routes/metadata/index.ts

```diff
@@ -65,241 +32,73 @@ export const endpointFilters = schema.object({
-  metadataRequestContext: MetadataRequestContext,
-  hostMetadata: HostMetadata
-): Promise<Agent | undefined> {
-  try {
-    return await metadataRequestContext.agentService.getAgent(
-      metadataRequestContext.requestHandlerContext.core.savedObjects.client,
-      hostMetadata.elastic.agent.id
-    );
-  } catch (e) {
-    if (
-      metadataRequestContext.requestHandlerContext.core.savedObjects.client.errors.isNotFoundError(
-        e
-      )
-    ) {
-      metadataRequestContext.logger.warn(
-        `agent with id ${hostMetadata.elastic.agent.id} not found`
-      );
-      return undefined;
-    } else {
-      throw e;
-    }
-  }
-}
 
-async function mapToHostResultList(
-  // eslint-disable-next-line @typescript-eslint/no-explicit-any
-  queryParams: Record<string, any>,
-  searchResponse: SearchResponse<HostMetadataDetails>,
-  metadataRequestContext: MetadataRequestContext
-): Promise<HostResultList> {
-  const totalNumberOfHosts =
-    ((searchResponse.hits?.total as unknown) as { value: number; relation: string }).value || 0;
-  if (searchResponse.hits.hits.length > 0) {
-    return {
-      request_page_size: queryParams.size,
-      request_page_index: queryParams.from,
-      hosts: await Promise.all(
-        searchResponse.hits.hits.map(async (entry) =>
-          enrichHostMetadata(entry._source.HostDetails, metadataRequestContext)
-        )
-      ),
-      total: totalNumberOfHosts,
-    };
-  } else {
-    return {
-      request_page_size: queryParams.size,
-      request_page_index: queryParams.from,
-      total: totalNumberOfHosts,
-      hosts: [],
-    };
-  }
-}
-
-async function enrichHostMetadata(
-  hostMetadata: HostMetadata,
-  metadataRequestContext: MetadataRequestContext
-): Promise<HostInfo> {
-  let hostStatus = HostStatus.ERROR;
-  let elasticAgentId = hostMetadata?.elastic?.agent?.id;
-  const log = metadataRequestContext.logger;
-  try {
-    /**
-     * Get agent status by elastic agent id if available or use the host id.
-     */
-
-    if (!elasticAgentId) {
-      elasticAgentId = hostMetadata.host.id;
-      log.warn(`Missing elastic agent id, using host id instead ${elasticAgentId}`);
-    }
-
-    const status = await metadataRequestContext.agentService.getAgentStatusById(
-      metadataRequestContext.requestHandlerContext.core.savedObjects.client,
-      elasticAgentId
-    );
-    hostStatus = HOST_STATUS_MAPPING.get(status) || HostStatus.ERROR;
-  } catch (e) {
-    if (
-      metadataRequestContext.requestHandlerContext.core.savedObjects.client.errors.isNotFoundError(
-        e
-      )
-    ) {
-      log.warn(`agent with id ${elasticAgentId} not found`);
-    } else {
-      log.error(e);
-      throw e;
-    }
-  }
-  return {
-    metadata: hostMetadata,
-    host_status: hostStatus,
-  };
+  router.get(
+    {
+      path: `${GET_METADATA_REQUEST_ROUTE}`,
+      validate: GetMetadataRequestSchema,
+      options: { authRequired: true, tags: ['access:securitySolution'] },
```

Comment:
Q. What does the `tags` do?

Because I may have missed that in the Trusted Apps APIs. 🙂

Reply:
it is for space access control